### DISCUSSION TITLE
Support for custom colors for different components

### DISCUSSION THREAD

**Discussion by AyushSehrawat** [squidfunk/mkdocs-material#4716] [Q&A] [answered]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

Hi! I would like if you were to add some option edit the colors of some component, because when i tried with extra.css ( custom stylesheet ), some things didn't work the way i intended, and at some places it wasn't even taking part, like overwritten. Also is there any guide on components you use ? Like `md-header`, `h1,h2,....`...

### Use Cases

Custom color for better flavouring of website. It will give opportunities to more varieties of themes, instead of being restricted to some few. 
Why? Like i added an a:hover and `md-nav__link`, but it didn't work everywhere. Only at some places.

**Comment by squidfunk** [comment, marked as answer]
Thanks for suggesting. As you have noticed, appearance can be changed and customized with [additional CSS](https://squidfunk.github.io/mkdocs-material/customization/#additional-css). Unfortunately, we can't provide a general web development guide in our documentation, detailing for every component how to override it. This is out-of-scope for the project. I'm converting this to a discussion. Feel free to share what you tried, maybe somebody can help 😊

**Comment by AyushSehrawat** [comment]
@squidfunk is it possible to get some name of core component like just links, text because the extra.css only seems to edit the middle component of screen, i.e the one with content. Not the side bars on each side (If enabled). Some quick guide to tweak them would be helpful.

**Reply by squidfunk** [reply]
You can edit everything with additional CSS, not only the content. If we would start a guide, people would be asking for more and more guidance on each and every component. I recommend searching the issue tracker and discussion board for customizations.

**Reply by AyushSehrawat** [reply]
Oh i can see that coming if you start a guide... I will look through them! Thank you :)

**Comment by AyushSehrawat** [comment]
I was trying to implement some colors of https://github.com/catppuccin/catppuccin.. Though failed to edit background color because for some weird reasons, it showed shadow under heading of sidebars

**Reply by squidfunk** [reply]
For colors, see our [custom colors guide](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#custom-colors) and [color definitions](https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/stylesheets/main/_colors.scss):

```css
:root > * {
  --md-primary-fg-color:        #EE0F0F;
  --md-primary-fg-color--light: #ECB7B7;
  --md-primary-fg-color--dark:  #90030C;
}
```

**Reply by AyushSehrawat** [reply]
This would be enough for my use case. Thanks a lot.
